File: Cpp/1897_Redistribute_Characters_to_Make_All_Strings_Equal.cpp
class Solution {
public:
    bool makeEqual(vector<string>& words) {
        vector<int> frequency(26, 0);
        
        for(auto& word : words) {
            for(auto& ch : word) {
                ++frequency[ch - 'a'];
            }
        }
        
        for(auto& it : frequency) {
            if(it % words.size() != 0) return false;
        }
        
        return true;
    }
};

File: Kotlin/1897_Redistribute_Characters_to_Make_All_Strings_Equal.kt
class Solution {
    fun makeEqual(words: Array<String>): Boolean {
        val frequency = IntArray(26)

        words.forEach { word ->
            word.forEach { ch ->
                ++frequency[ch - 'a']
            }
        }
                        
        for(it in frequency) {
            if(it % words.size != 0) return false
        }
        
        return true
    }
}